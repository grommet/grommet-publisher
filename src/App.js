import React from 'react';
import {
  Box,
  Grid,
  Grommet,
  ResponsiveContext,
  Keyboard,
  grommet,
} from 'grommet';
import { Document } from 'grommet-icons';
import { apiUrl, starter, upgradeSite, publish } from './site';
import Router from './Router';
import Nav from './Nav/Nav';
import Editor from './Editor';
import Preview from './Preview';

const getParams = () => {
  const { location } = window;
  const params = {};
  location.search
    .slice(1)
    .split('&')
    .forEach(p => {
      const [k, v] = p.split('=');
      params[k] = decodeURIComponent(v);
    });
  return params;
};

const App = () => {
  const [site, setSite] = React.useState();
  const [preview, setPreview] = React.useState();
  const [changes, setChanges] = React.useState([]);
  const [changeIndex, setChangeIndex] = React.useState();
  const storeTimer = React.useRef(null);

  // load initial site
  React.useEffect(() => {
    const params = getParams();
    if (params.id) {
      fetch(`${apiUrl}/${params.id}`)
        .then(response => response.json())
        .then(nextSite => {
          if (!nextSite.id) nextSite.id = params.id;
          upgradeSite(nextSite);
          document.title = nextSite.name;
          setPreview(true);
          setSite(nextSite);
          setChanges([{ site: nextSite }]);
          // save it for future offline usage
          localStorage.setItem(params.id, JSON.stringify(nextSite));
        })
        .catch(() => {
          // see if we have this site cached, we might be offline
          const stored = localStorage.getItem(params.id);
          if (stored) {
            const nextSite = JSON.parse(stored);
            upgradeSite(nextSite);
            document.title = nextSite.name;
            setPreview(true);
            setSite(nextSite);
            setChanges([{ site: nextSite }]);
          }
        });
    } else {
      let stored = localStorage.getItem('activeSite');
      if (stored) {
        stored = localStorage.getItem(stored);
      }
      if (stored) {
        const nextSite = JSON.parse(stored);
        upgradeSite(nextSite);
        document.title = nextSite.name;
        setSite(nextSite);
        setChanges([{ site: nextSite }]);
      } else {
        setSite(starter);
        setChanges([{ site: starter }]);
      }
    }
  }, []);

  // add to changes, if needed
  React.useEffect(() => {
    // do this stuff lazily to ride out typing sprees
    const timer = setTimeout(() => {
      // If we already have this site object, we must be doing an undo or
      // redo, and therefore no need to add a change
      if (!changes.some(c => c.site === site)) {
        let nextChanges;
        nextChanges = [...changes];
        nextChanges = nextChanges.slice(changeIndex, 10);
        nextChanges.unshift({ site });
        setChanges(nextChanges);
        setChangeIndex(0);
      }
    }, 1000);
    return () => clearTimeout(timer);
  }, [changes, changeIndex, site]);

  const onUndo = React.useCallback(() => {
    const nextChangeIndex = Math.min(changeIndex + 1, changes.length - 1);
    const { site: nextSite } = changes[nextChangeIndex];
    setSite(nextSite);
    setChangeIndex(nextChangeIndex);
  }, [changes, changeIndex]);

  const onRedo = React.useCallback(() => {
    const nextChangeIndex = Math.max(changeIndex - 1, 0);
    const { site: nextSite } = changes[nextChangeIndex];
    setSite(nextSite);
    setChangeIndex(nextChangeIndex);
  }, [changes, changeIndex]);

  const onChange = nextSite => {
    setSite(nextSite);

    // delay storing it locally so we don't bog down typing
    clearTimeout(storeTimer.current);
    storeTimer.current = setTimeout(() => {
      document.title = nextSite.name;
      localStorage.setItem(nextSite.name, JSON.stringify(nextSite));
      localStorage.setItem('activeSite', nextSite.name);
      if (document.location.search) {
        // clear current URL, in case we've started editing a published design locally
        window.history.replaceState({}, nextSite.name, '/');
      }
      // ensure we have this site name in our site list
      const sites = JSON.parse(localStorage.getItem('sites') || '[]');
      if (!sites.includes(nextSite.name)) {
        sites.unshift(nextSite.name);
        localStorage.setItem('sites', JSON.stringify(sites));
      }
    }, 1000);
  };

  const onKey = event => {
    if (event.metaKey) {
      if (event.key === 'e') {
        event.preventDefault();
        setPreview(!preview);
      } else if (event.key === 'p' && event.shiftKey) {
        const stored = localStorage.getItem('identity');
        if (stored) {
          const identity = JSON.parse(stored);
          publish({
            site,
            ...identity,
            onChange,
            onError: error => console.error(error),
          });
        } else {
          console.warn('You need to have published to be able to re-publish');
        }
      }
    }
  };

  return (
    <Router>
      <Grommet theme={grommet} style={{ minHeight: '100vh' }}>
        <Keyboard target="document" onKeyDown={onKey}>
          <ResponsiveContext.Consumer>
            {responsive => {
              return !site ? (
                <Box fill justify="center" align="center" margin="xlarge">
                  <Box animation="pulse">
                    <Document size="xlarge" />
                  </Box>
                </Box>
              ) : (
                <Grid
                  fill
                  columns={
                    responsive === 'small' || preview
                      ? 'flex'
                      : ['small', ['small', 'large'], ['large', 'flex']]
                  }
                  rows="full"
                >
                  {responsive !== 'small' && !preview && (
                    <Nav
                      site={site}
                      onChange={onChange}
                      onRedo={changeIndex > 0 && onRedo}
                      onUndo={changeIndex < changes.length - 1 && onUndo}
                    />
                  )}
                  {responsive !== 'small' && !preview && (
                    <Editor site={site} onChange={onChange} />
                  )}
                  <Preview site={site} onChange={onChange} />
                </Grid>
              );
            }}
          </ResponsiveContext.Consumer>
        </Keyboard>
      </Grommet>
    </Router>
  );
};

export default App;
