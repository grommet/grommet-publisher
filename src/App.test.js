import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import App from './App';

describe('App', () => {
  afterEach(cleanup);

  test('empty', () => {
    const { getByTitle, container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(getByTitle('choose another site'));
    expect(container.firstChild).toMatchSnapshot();
  });
});
