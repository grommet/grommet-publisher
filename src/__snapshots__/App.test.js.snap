// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App empty 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 gVQrIe"
  style="min-height: 100vh;"
>
  <div
    class="StyledGrid-sc-1wofa1l-0 jtEBxd"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jPhZMU"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fiFliE"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 hzPhWp"
            title="choose another site"
            type="button"
          >
            <svg
              aria-label="Apps"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,5 L21,5 L21,3 L19,3 L19,5 Z M11,5 L13,5 L13,3 L11,3 L11,5 Z M3,5 L5,5 L5,3 L3,3 L3,5 Z M19,13 L21,13 L21,11 L19,11 L19,13 Z M11,13 L13,13 L13,11 L11,11 L11,13 Z M3,13 L5,13 L5,11 L3,11 L3,13 Z M19,21 L21,21 L21,19 L19,19 L19,21 Z M11,21 L13,21 L13,19 L11,19 L11,21 Z M3,21 L5,21 L5,19 L3,19 L3,21 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 jeaTYv"
            disabled=""
            title="undo last change"
            type="button"
          >
            <svg
              aria-label="Undo"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.18,4,8.6,5.44,6.06,8h9.71a6,6,0,0,1,0,12h-2V18h2a4,4,0,0,0,0-8H6.06L8.6,12.51,7.18,13.92,2.23,9Z"
              />
            </svg>
          </button>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 jeaTYv"
            disabled=""
            title="redo last change"
            type="button"
          >
            <svg
              aria-label="Redo"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.82,4,15.4,5.44,17.94,8H8.23a6,6,0,0,0,0,12h2V18h-2a4,4,0,0,1,0-8h9.71L15.4,12.51l1.41,1.41L21.77,9Z"
              />
            </svg>
          </button>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 hzPhWp"
            type="button"
          >
            <svg
              aria-label="Share"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,13 L19,23 L1,23 L1,5 L1,5 L11,5 M14,1 L23,1 L23,10 M10,14 L22.9999998,1 L10,14 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 hTZIev"
      >
        <a
          class="StyledButton-sc-323bzc-0 cImFMH"
          href="/"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 ekijOn"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gWIGcR"
            >
              My Site
            </span>
          </div>
        </a>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <a
            class="StyledButton-sc-323bzc-0 iElqND"
            href="/welcome"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 koNkem"
            >
              <span
                class="StyledText-sc-1sadyjn-0 eGyAkH"
              >
                welcome
              </span>
            </div>
          </a>
          <div
            class="StyledBox-sc-13pk1d4-0 ksdjXW"
          >
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/join"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  join
                </span>
              </div>
            </a>
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/explore"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  explore
                </span>
              </div>
            </a>
            <button
              class="StyledButton-sc-323bzc-0 iElqND"
              title="add a page"
              type="button"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jJvHJM"
              >
                <svg
                  aria-label="FormAdd"
                  class="StyledIcon-ofa7kd-0 hozXcj"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,18 L12,6 M6,12 L18,12"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <a
            class="StyledButton-sc-323bzc-0 iElqND"
            href="/about"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 koNkem"
            >
              <span
                class="StyledText-sc-1sadyjn-0 eGyAkH"
              >
                about
              </span>
            </div>
          </a>
          <div
            class="StyledBox-sc-13pk1d4-0 ksdjXW"
          >
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/contact"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  contact
                </span>
              </div>
            </a>
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/legal"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  legal
                </span>
              </div>
            </a>
            <button
              class="StyledButton-sc-323bzc-0 iElqND"
              title="add a page"
              type="button"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jJvHJM"
              >
                <svg
                  aria-label="FormAdd"
                  class="StyledIcon-ofa7kd-0 hozXcj"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,18 L12,6 M6,12 L18,12"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <button
          class="StyledButton-sc-323bzc-0 iElqND"
          title="add a chapter"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 jJvHJM"
          >
            <svg
              aria-label="FormAdd"
              class="StyledIcon-ofa7kd-0 hozXcj"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,18 L12,6 M6,12 L18,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 jyWDiZ"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bHorkR"
      >
        <button
          class="StyledButton-sc-323bzc-0 iElqND"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 gersVE"
          >
            content
          </div>
        </button>
        <button
          class="StyledButton-sc-323bzc-0 iElqND"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eXTSqT"
          >
            details
          </div>
        </button>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 bimCvx"
      >
        <textarea
          class="StyledTextArea-sc-17i3mwp-0 fUMFqD"
          id="content"
          name="content"
        >
          # My Site
        </textarea>
      </div>
    </div>
    <div
      class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
      style="min-height: 100vh;"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 lcBqZl"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 jdsUxI"
        >
          <svg
            aria-label="Document"
            class="StyledIcon-ofa7kd-0 IKQZf"
            viewBox="0 0 24 24"
          >
            <path
              d="M14,1 L14,8 L21,8 M21,23 L3,23 L3,1 L15,1 L18,4 L21,7 L21,23 L21,23 L21,23 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App empty 2`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 gVQrIe"
  style="min-height: 100vh;"
>
  <div
    class="StyledGrid-sc-1wofa1l-0 jtEBxd"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jPhZMU"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fiFliE"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 hzPhWp"
            title="choose another site"
            type="button"
          >
            <svg
              aria-label="Apps"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,5 L21,5 L21,3 L19,3 L19,5 Z M11,5 L13,5 L13,3 L11,3 L11,5 Z M3,5 L5,5 L5,3 L3,3 L3,5 Z M19,13 L21,13 L21,11 L19,11 L19,13 Z M11,13 L13,13 L13,11 L11,11 L11,13 Z M3,13 L5,13 L5,11 L3,11 L3,13 Z M19,21 L21,21 L21,19 L19,19 L19,21 Z M11,21 L13,21 L13,19 L11,19 L11,21 Z M3,21 L5,21 L5,19 L3,19 L3,21 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 jeaTYv"
            disabled=""
            title="undo last change"
            type="button"
          >
            <svg
              aria-label="Undo"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.18,4,8.6,5.44,6.06,8h9.71a6,6,0,0,1,0,12h-2V18h2a4,4,0,0,0,0-8H6.06L8.6,12.51,7.18,13.92,2.23,9Z"
              />
            </svg>
          </button>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 jeaTYv"
            disabled=""
            title="redo last change"
            type="button"
          >
            <svg
              aria-label="Redo"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.82,4,15.4,5.44,17.94,8H8.23a6,6,0,0,0,0,12h2V18h-2a4,4,0,0,1,0-8h9.71L15.4,12.51l1.41,1.41L21.77,9Z"
              />
            </svg>
          </button>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <button
            class="StyledButton-sc-323bzc-0 hzPhWp"
            type="button"
          >
            <svg
              aria-label="Share"
              class="StyledIcon-ofa7kd-0 kPmYvD"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,13 L19,23 L1,23 L1,5 L1,5 L11,5 M14,1 L23,1 L23,10 M10,14 L22.9999998,1 L10,14 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 hTZIev"
      >
        <a
          class="StyledButton-sc-323bzc-0 cImFMH"
          href="/"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 ekijOn"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gWIGcR"
            >
              My Site
            </span>
          </div>
        </a>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <a
            class="StyledButton-sc-323bzc-0 iElqND"
            href="/welcome"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 koNkem"
            >
              <span
                class="StyledText-sc-1sadyjn-0 eGyAkH"
              >
                welcome
              </span>
            </div>
          </a>
          <div
            class="StyledBox-sc-13pk1d4-0 ksdjXW"
          >
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/join"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  join
                </span>
              </div>
            </a>
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/explore"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  explore
                </span>
              </div>
            </a>
            <button
              class="StyledButton-sc-323bzc-0 iElqND"
              title="add a page"
              type="button"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jJvHJM"
              >
                <svg
                  aria-label="FormAdd"
                  class="StyledIcon-ofa7kd-0 hozXcj"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,18 L12,6 M6,12 L18,12"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hTZIev"
        >
          <a
            class="StyledButton-sc-323bzc-0 iElqND"
            href="/about"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 koNkem"
            >
              <span
                class="StyledText-sc-1sadyjn-0 eGyAkH"
              >
                about
              </span>
            </div>
          </a>
          <div
            class="StyledBox-sc-13pk1d4-0 ksdjXW"
          >
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/contact"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  contact
                </span>
              </div>
            </a>
            <a
              class="StyledButton-sc-323bzc-0 iElqND"
              href="/legal"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 koNkem"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eGyAkH"
                >
                  legal
                </span>
              </div>
            </a>
            <button
              class="StyledButton-sc-323bzc-0 iElqND"
              title="add a page"
              type="button"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jJvHJM"
              >
                <svg
                  aria-label="FormAdd"
                  class="StyledIcon-ofa7kd-0 hozXcj"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,18 L12,6 M6,12 L18,12"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <button
          class="StyledButton-sc-323bzc-0 iElqND"
          title="add a chapter"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 jJvHJM"
          >
            <svg
              aria-label="FormAdd"
              class="StyledIcon-ofa7kd-0 hozXcj"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,18 L12,6 M6,12 L18,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 jyWDiZ"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bHorkR"
      >
        <button
          class="StyledButton-sc-323bzc-0 iElqND"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 gersVE"
          >
            content
          </div>
        </button>
        <button
          class="StyledButton-sc-323bzc-0 iElqND"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eXTSqT"
          >
            details
          </div>
        </button>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 bimCvx"
      >
        <textarea
          class="StyledTextArea-sc-17i3mwp-0 fUMFqD"
          id="content"
          name="content"
        >
          # My Site
        </textarea>
      </div>
    </div>
    <div
      class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
      style="min-height: 100vh;"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 lcBqZl"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 jdsUxI"
        >
          <svg
            aria-label="Document"
            class="StyledIcon-ofa7kd-0 IKQZf"
            viewBox="0 0 24 24"
          >
            <path
              d="M14,1 L14,8 L21,8 M21,23 L3,23 L3,1 L15,1 L18,4 L21,7 L21,23 L21,23 L21,23 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
